ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -352,14 +357,18 @@ func (r *RayDashboardClient) GetJobInfo(ctx context.Context, jobId string) (*Ray
 	defer resp.Body.Close()
 
 	if resp.StatusCode == http.StatusNotFound {
-		return nil, nil
+		return nil, ErrJobInfoNotFound
```

Comment:
Hi @kingeasternsun,

I noticed that this `if resp.StatusCode == http.StatusNotFound` check should be moved to after the below `io.ReadAll(resp.Body)`

Could you also help correct that? Thanks!

Reply:
Thanks for your review, I will fix that.

Reply:
@rueian    this place  `io.ReadAll(resp.Body)`  should I check the return error as before, or  changed to  `body, _ := io.ReadAll(resp.Body)` like other methods below?

Reply:
I think the current implementation is ok. No need to change.

Reply:
OK